### DISCUSSION TITLE
Librocket Backwards Compatibility

### DISCUSSION THREAD

**Discussion by MjnMixael** [mikke89/RmlUi#665] [General]
Hello. I work on a game engine called FreespaceOpen ([source on Github here](https://github.com/scp-fs2open/fs2open.github.com)) and several years ago we added the librocket library. To my recollection, the person who did this work did have to [make a few small custom changes to the library](https://github.com/scp-fs2open/libRocket/tree/ecd648a43aff8a9f3daf064d75ca5725237d5b38) but overall it seems to follow the librocket core documentation exactly.

Now, librocket is old and not maintained anymore. You all know this of course. That's how I ended up here. I'm wondering how easy it would be to swap out our librocket library for rmlGUI. Ignoring our custom changes for the moment.. how backwards compatible is rmlGUI compared to the old librocket library? Are there any points of big breaking changes to be aware of?

**Comment by MjnMixael** [comment]
Bah... just after I posed this I found the full changelog with a Breaking Changes section. Nevermind!

**Reply by mikke89** [reply]
Hey, and indeed, I've tried to document breaking changes in detail for every release. Hopefully the transition goes relatively smooth if you do decide to go ahead with it. Good luck!
